include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -0,0 +1,1061 @@
+                n = n * n + 2 * negative; // for switch case order
+
+                switch (n)
+                {
+                    case 0:
+                    {
+                        result[result.size() - 1] = 'C';
+                        result.append(reinterpret_cast<char const *>(&arg), 1);
+                        break;
+                    }
+                    case 1:
+                    {
+                        result[result.size() - 1] = 'S';
+                        result.append(reinterpret_cast<char const *>(&arg), 2);
+                        break;
+                    }
+                    case 2:
+                    {
+                        result[result.size() - 1] = 'c';
+                        int8_t tmp = static_cast<int8_t>(arg);
+                        result.append(reinterpret_cast<char const *>(&tmp), 1);
+                        break;
+                    }
+                    case 3:
+                    {
+                        result[result.size() - 1] = 's';
+                        int16_t tmp = static_cast<int16_t>(arg);
+                        result.append(reinterpret_cast<char const *>(&tmp), 2);
+                        break;
+                    }
+                    default:
+                    {
+                        result.append(reinterpret_cast<char const *>(&arg), 4); // always i
+                        break;
+                    }
+                }
+            }
+            else if constexpr (std::Same<T, std::string>)
+            {
+                result.append(reinterpret_cast<char const *>(arg.data()), arg.size() + 1/*+ null character*/);
+            }
+            else if constexpr (!std::ranges::Range<T>) // char, float
+            {
+                result.append(reinterpret_cast<char const *>(&arg), sizeof(arg));
+            }
+            else // std::vector of some Arithmetic_type type
+            {
+                int32_t sz{static_cast<int32_t>(arg.size())};
+                result.append(reinterpret_cast<char *>(&sz), 4);
+                result.append(reinterpret_cast<char const *>(arg.data()), arg.size() * sizeof(value_type_t<T>));
+            }
+        };
+
+        for (auto & [tag, variant] : tag_dict)
+        {
+            result.push_back(static_cast<char>(tag / 256));
+            result.push_back(static_cast<char>(tag % 256));
+
+            result.push_back(detail::sam_tag_type_char[variant.index()]);
+
+            if (!is_char<'\0'>(detail::sam_tag_type_char_extra[variant.index()]))
+                result.push_back(detail::sam_tag_type_char_extra[variant.index()]);
+
+            std::visit(stream_variant_fn, variant);
+        }
+
+        return std::move(result);
+    }
+
+    //!\brief Computes the bin number for a given region [beg, end), copied from the official SAM specifications.
+    static uint16_t reg2bin(int32_t beg, int32_t end) noexcept
```

Comment:
Why is this static and others not?

Reply:
the read_field function won;t be able to be static soon anyway when me move the stream handle into the format.